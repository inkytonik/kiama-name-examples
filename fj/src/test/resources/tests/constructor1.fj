class A extends Object {
  Object f;
  A(Object g) {
    super();
    this.f = g;
  }
}
