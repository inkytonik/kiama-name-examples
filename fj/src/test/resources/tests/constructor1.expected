src/test/resources/tests/constructor1.fj:2:10:info: f def: Object
  Object f;
         ^
src/test/resources/tests/constructor1.fj:3:3:error: constructor args (Object g) should be the fields (Object f)
  A(Object g) {
  ^
src/test/resources/tests/constructor1.fj:3:12:info: g def: Object
  A(Object g) {
           ^
src/test/resources/tests/constructor1.fj:5:5:error: initialiser must be this.f = f
    this.f = g;
    ^
